src/org/labkey/test/util/ProcessHelper.java

```diff
@@ -0,0 +1,140 @@
+package org.labkey.test.util;
+
+import org.apache.commons.lang3.ArrayUtils;
+import org.jetbrains.annotations.NotNull;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.nio.charset.Charset;
+import java.time.Duration;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+
+public class ProcessHelper
+{
+    private final ProcessBuilder pb;
+    private Duration timeout = Duration.ofSeconds(60);
+
+    public ProcessHelper(String executable, String... args)
+    {
+        pb = new ProcessBuilder();
+        pb.command(ArrayUtils.addAll(new String[]{executable}, args));
+        pb.redirectErrorStream(true);
+    }
+
+    public ProcessHelper(File executable, String... args)
+    {
+        this(executable.toPath().normalize().toAbsolutePath().toString(), args);
+    }
+
+    public Map<String, String> environment()
+    {
+        return pb.environment();
+    }
+
+    public ProcessHelper setTimeout(@NotNull Duration timeout)
+    {
+        this.timeout = timeout;
+        return this;
+    }
+
+    public String getProcessOutput() throws IOException
+    {
+        return getProcessOutput(false);
+    }
+
+    public String getProcessOutput(boolean logOutput) throws IOException
+    {
+        StringBuffer output = new StringBuffer();
+        runProcess(output, logOutput);
+        return output.toString();
+    }
+
+    /**
+     * Adapted from `org.labkey.api.reports.ExternalScriptEngine#runProcess()`
+     * Execute the external process in separate thread
+     * @return the exit code for the invocation - 0 if the process completed successfully.
+     */
```

Comment:
This looks ok to me (one question below) but I'm depending upon the comment above that this is adapted from org.labkey.api. A well respected and trusted source of high quality software!